exercises/practice/diamond/diamond_test.rb

```diff
@@ -2,34 +2,105 @@
 require_relative 'diamond'
 
 class DiamondTest < Minitest::Test
-  def test_letter_a
+  def test_degenerate_case_with_a_single_a_row
+    # skip
     answer = Diamond.make_diamond('A')
-    assert_equal "A\n", answer
+    string = "A\n"\
+
+    assert_equal string, answer
+  end
+
+  def test_degenerate_case_with_no_row_containing_3_distinct_groups_of_spaces
+    skip
+    answer = Diamond.make_diamond('B')
+    string = " A \n"\
+            "B B\n"\
+            " A \n"\
+
+    assert_equal string, answer
   end
 
-  def test_letter_c
+  def test_smallest_non_degenerate_case_with_odd_diamond_side_length
     skip
     answer = Diamond.make_diamond('C')
     string = "  A  \n"\
-             " B B \n"\
-             "C   C\n"\
-             " B B \n"\
-             "  A  \n"
+            " B B \n"\
+            "C   C\n"\
+            " B B \n"\
+            "  A  \n"\
+
     assert_equal string, answer
   end
 
-  def test_letter_e
+  def test_smallest_non_degenerate_case_with_even_diamond_side_length
     skip
-    answer = Diamond.make_diamond('E')
-    string = "    A    \n"\
-             "   B B   \n"\
-             "  C   C  \n"\
-             " D     D \n"\
-             "E       E\n"\
-             " D     D \n"\
-             "  C   C  \n"\
-             "   B B   \n"\
-             "    A    \n"
+    answer = Diamond.make_diamond('D')
+    string = "   A   \n"\
+            "  B B  \n"\
+            " C   C \n"\
+            "D     D\n"\
+            " C   C \n"\
+            "  B B  \n"\
+            "   A   \n"\
+
+    assert_equal string, answer
+  end
+
+  def test_largest_possible_diamond
+    skip
+    answer = Diamond.make_diamond('Z')
+    string = "                         A                         \n"\
```

Comment:
Consider using HereDoc for this, rather than consecutive string "trick".

Reply:
Same for the others.

Reply:
I tested it out but the problem I find is that for this exercise is trailing spaces important. So what end up happening is that each line is appended with #{'       '}. Which in my opinion, looks just weird.